lib/new_relic/agent/instrumentation/net_http/instrumentation.rb

```diff
@@ -32,12 +32,28 @@ def request_with_tracing(request)
             end
 
             wrapped_response = NewRelic::Agent::HTTPClients::NetHTTPResponse.new(response)
+
+            if openai_parent?(segment)
+              populate_openai_response_headers(wrapped_response, segment.parent)
+            end
+
             segment.process_response_headers(wrapped_response)
+
             response
           ensure
             segment&.finish
           end
         end
+
+        def openai_parent?(segment)
+          segment&.parent&.name&.match?(/Llm\/.*\/OpenAI\/.*/)
```

Comment:
We should freeze the pattern and keep reusing the same frozen object. We can also use `%r{}` to prevent having to escape the forward slashes.

```diff
- segment&.parent&.name&.match?(/Llm\/.*\/OpenAI\/.*/)

+ OPENAI_SEGMENT_PATTERN = %r{Llm/.*/OpenAI/.*}.freeze
+
+ segment&.parent&.name&.match?(OPENAI_SEGMENT_PATTERN)
```

Reply:
Learning moment! What is the benefit of freezing the pattern/creating a new object here if we're only doing this type of check once?

Reply:
@hannahramadan From what I can tell, the `openai_parent?` method gets invoked once for every single wrapped `Net::HTTP` call made to check to see if OpenAI was involved or not. A Ruby server running in a loop could potentially perform thousands of `Net::HTTP` calls - especially for non AI use cases - and every call would cause Ruby to spawn a fresh object to hold the same regex pattern over and over again.

Reply:
Thank you @fallwith! That makes a lot of sense. I've updated this: https://github.com/newrelic/newrelic-ruby-agent/pull/2398/commits/d43b1a6bb130f181b3b7a05c41d04b3cb219afb3